src/Parse/Grammar.y

```diff
@@ -61,21 +63,24 @@ import           Explain.Meta
   '||'     { Token _ TokenDoublePipe }
   '>'      { Token _ TokenRightChevron }
   '<'      { Token _ TokenLeftChevron }
+  ')>'     { Token _ TokenRightParenChevron }
+  '<('     { Token _ TokenLeftChevronParen }
   '>='     { Token _ TokenRightChevronEq }
   '<='     { Token _ TokenLeftChevronEq }
   '!'      { Token _ TokenExclamationMark }
 
 
-%left '?' '->' '|' where is 'if'
+%nonassoc LOWEST
+%left '?' '->' '|' where is 'if' '='
 %left ':' 'else'
 %left '|>'
 %left '>' '<' '>=' '<=' '=='
 %left '+' '-' '||'
-%left '*' '/' '&&'
+%left '*' '/' '%' '&&'
 %left ','
-%nonassoc '(' ')' '=' '=>' '::' where is 'ret' '{' '}' '[' ']'
+%nonassoc '(' ')' '<(' ')>' '=>' '::' where is 'ret' '{' '}' '[' ']'
 %right '!'
-%left HIGHEST
+%nonassoc HIGHEST
```

Comment:
Interesting. Where do `LOWEST` / `HIGHEST` come from?

Reply:
Oh, it's names you can define and use %prec to override precedence of some terminal tokens. But I figured it only works if that if it is used at the beginning of a production. So mostly unused. That's a leftover from my attempts for tuple grammar. Should be deleted.